imbi/endpoints/project_activity_feed.py

```diff
@@ -1,57 +1,116 @@
-                       a.recorded_by AS who,
-                       'updated fact' AS what,
-                       b.name AS fact_name,
-                       a.value
-                  FROM v1.project_facts AS a
-                  JOIN v1.project_fact_types AS b
-                    ON b.id = a.fact_type_id
-                 WHERE a.project_id = %(project_id)s),
-            combined AS (
-                SELECT *
-                  FROM created
-                 UNION
-                SELECT *
-                  FROM updated
-                 UNION
-                SELECT *
-                  FROM facts)
-        SELECT a.when,
-               a.who,
-               b.display_name,
-               a.what,
-               a.fact_name,
-               a.value,
-               'ProjectFeedEntry' as "type"
-          FROM combined AS a
-          JOIN v1.users AS b
-            ON b.username = a.who
-         ORDER BY a.when DESC;""")
+          SELECT created_at AS "when", created_by AS who,
+                 'created' AS what, NULL AS fact_name, NULL AS value
+            FROM v1.projects
+           WHERE id = %(project_id)s
+        ),
+        updated AS (
+          SELECT last_modified_at AS "when", last_modified_by AS who,
+                 'updated' AS what, NULL AS fact_name, NULL AS value
+            FROM v1.projects
+           WHERE id = %(project_id)s
+        ),
+        facts AS (
+          SELECT a.recorded_at AS "when", a.recorded_by AS who,
+                 'updated fact' AS what, b.name AS fact_name, a.value
+            FROM v1.project_facts AS a
+            JOIN v1.project_fact_types AS b
+              ON b.id = a.fact_type_id
+           WHERE a.project_id = %(project_id)s
+        ),
+        combined AS (
+                SELECT * FROM created
+          UNION SELECT * FROM updated
+          UNION SELECT * FROM facts
+        )
+        SELECT 'ProjectFeedEntry' AS "type", c."when", c.who, u.display_name,
+               c.what, c.fact_name, c.value
+          FROM combined AS c
+          JOIN v1.users AS u ON u.username = c.who
+         WHERE c."when" >  %(earlier)s
+           AND c."when" <= %(later)s
+         ORDER BY c."when" DESC
+         LIMIT %(remaining)s
+        """)
+
+    OPERATIONS_LOG_SQL = re.sub(
+        r'\s+', ' ', """\
+        SELECT 'OperationsLogEntry' AS "type", o.id,
+               o.recorded_at, o.recorded_by, o.completed_at,
+               o.project_id, o.environment, o.change_type,
+               o.description, o.link, o.notes, o.ticket_slug,
+               o.version, p.name AS project_name, u.email_address,
+               u.display_name
+          FROM v1.operations_log AS o
+          LEFT JOIN v1.projects AS p ON p.id = o.project_id
+          LEFT JOIN v1.users AS u ON u.username = o.recorded_by
+         WHERE recorded_at >  %(earlier)s
+           AND recorded_at <= %(later)s
+         ORDER BY o.recorded_at DESC, o.id DESC
+         LIMIT %(remaining)s
+        """)
```

Comment:
We need to filter by `project_id` here.

Reply:
hmm... yes... yes we do.  I can also drop the `LEFT` in the `LEFT JOIN` against `v1.projects` since we only want operation log entries that are associated with a project.